### PR TITLE
Mapping between the existing API and the new data model

### LINKED ISSUE #21: Map Between API and New Data Model
## Description
Create a mapping between the existing API and the new data model to accommodate "Supplier" and "Customer" columns.

## Product Name
[vas-country-risk-backend](https://github.com/eclipse-tractusx/vas-country-risk-backend)

## Tasks
- [x] Define the mapping between the existing API and the new data model.
- [x] Implement the mapping in the code.

### PR DESCRIPTION
**Pull request by fabiodmota**
---
title: '(feat): Mapping between the existing API and the new data model'
---

## Description

This PR introduces a new feature that creates a mapping between the existing API and the new data model to accommodate "Supplier" and "Customer" columns. This is in response to the issue [#21](https://github.com/eclipse-tractusx/vas-country-risk-backend/issues/21) in the vas-country-risk-backend repository.

Fixes #21 

## Pre-review checks

- [x] DEPENDENCIES are up to date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] Copyright and license header are present on all affected files

### Additional information

- To request a review, check out [who's involved](https://projects.eclipse.org/projects/automotive.tractusx/who) to see a list of contributors and committers
- Use the [mailing list](https://accounts.eclipse.org/mailing-list/tractusx-dev) if you are unsure, who to contact, or if want to engage in a general discussion
- Check out our guide on [how to contribute](https://eclipse-tractusx.github.io/docs/oss/how-to-contribute)
- Check out our [Release Guidelines](https://eclipse-tractusx.github.io/docs/release)
- Check out the [Eclipse Foundation contribution guidelines](https://www.eclipse.org/projects/handbook/#contributing)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -49,6 +49,7 @@
 		<sonar.coverage.jacoco.xmlReportPaths>${project.reporting.outputDirectory}/target/jacoco-report/jacoco.xml</sonar.coverage.jacoco.xmlReportPaths>
 		<sonar.organization>catenax-ng</sonar.organization>
 		<sonar.projectKey>catenax-ng_tx-vas-country-risk-backend</sonar.projectKey>
+		<sonar.projectName>tx-vas-country-risk-backend</sonar.projectName>
```
Comment (SebastianBezold) [inline pom.xml:52]: Please prioritize eclipse-tractusx as leading repo. Therefore the naming should also follow Tractus-X. So maybe do not add the `tx-` prefix. You can add conditions to your workflows, that will configure the `sonar.projectName` via command parameter

**Review by SebastianBezold — APPROVED** [review]
Looks good from my side. Cannot evaluate the Java changes though

**Comment by fabiodmota** [comment]
@SebastianBezold thank already changed the workflow for the sonar properties and changed the pom.xml if all its ok for you please accept this merge

**Comment by alexsilva-CGI** [comment]
Tested application and it works fine, and Java code looks good. Ok from me

**Comment by SebastianBezold** [comment]
Hi @fabiodmota, since my last review you pushed new code to this PR. They are even annotated as feature.
The PR is now of a size, that requires an IP check. I'll create the issue for that and link it here. 
Also, the `pom.xml` changed quite a bit and to me it seems like the spring-boot dependencies are new. The `DEPENDENCIES` file on the other hand hardly changed. Can you clarify that? Did you run dash? Do we need to open automated IP issues for the 3rd party dependencies?

File: src/main/java/org/eclipse/tractusx/valueaddedservice/config/WebClientConfiguration.java
Comment (SebastianBezold) [inline src/main/java/org/eclipse/tractusx/valueaddedservice/config/WebClientConfiguration.java:1]: Copyright and License header missing

File: src/main/resources/config/application-dev.yml [hunk 2]
```diff
@@ -25,10 +25,16 @@ vas:
...
+      on-profile: dev
 
 
 application:
   name: dev
   host: http://project-value-added-service-vas/api/schedule
   partnersPoolUrl:
-    legalEntities: 'https://partners-pool.dev.demo.catena-x.net/api/catena/legal-entities'
\ No newline at end of file
+    bpdmAddressUrl: 'https://business-partners.dev.demo.catena-x.net/companies/test-company/api/catena/output/addresses/search?limit=10'
```
Comment (SebastianBezold) [inline src/main/resources/config/application-dev.yml:38]: Do you really need to specify this in the `application.yaml`? These are hard coded URLs to systems, that are not available for everyone. 
So please, if you are deploying your application, provide this as environment variables (spring boot picks that up for you) or any other option

File: src/main/resources/config/application-int.yml [hunk 3]
```diff
@@ -32,3 +32,8 @@ spring:
 
 application:
   name: int
+  host: http://project-value-added-service-vas/api/schedule
+  partnersPoolUrl:
+    bpdmAddressUrl: 'https://business-partners.int.demo.catena-x.net/companies/test-company/api/catena/output/addresses/search?limit=10'
```
Comment (SebastianBezold) [inline src/main/resources/config/application-int.yml:37]: See my other comment on hard coded config

File: src/test/java/org/eclipse/tractusx/valueaddedservice/service/logic/CountryLogicServiceTest.java [hunk 4]
```diff
@@ -51,23 +53,38 @@ class CountryLogicServiceTest {
     @InjectMocks
     CountryLogicService countryLogicService;
 
+    @Mock
+    ApplicationVariables applicationVariables;
+
```
Comment (SebastianBezold) [inline src/test/java/org/eclipse/tractusx/valueaddedservice/service/logic/CountryLogicServiceTest.java:58]: Unnecessary space

File: src/test/java/org/eclipse/tractusx/valueaddedservice/utils/DataBaseJsonConverterTest.java [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    @Test
+    public void convertToEntityAttributeWithNullJsonString() {
+        String customerInfoJSON = null;
+
+        Map<String, Object> result = dataBaseJsonConverter.convertToEntityAttribute(customerInfoJSON);
+
+        assertNull(result);
+    }
+
+
```
Comment (SebastianBezold) [inline src/test/java/org/eclipse/tractusx/valueaddedservice/utils/DataBaseJsonConverterTest.java:63]: Space?

File: src/test/resources/config/application.yml [hunk 6]
```diff
@@ -18,6 +18,19 @@
...
+vas:
+  datasource:
+    host: localhost
+    user: default
+    pass: defaultpassword
+  keycloakClientSecret: defautlSecret
+  keycloakClientId: country-risk-client
+  clientName: country-risk-client
+  authentication-url:
+    token-url: https://centralidp.dev.demo.catena-x.net/auth/realms/CX-Central/protocol/openid-connect/token
```
Comment (SebastianBezold) [inline src/test/resources/config/application.yml:30]: again hard coded config. To me it seems like a dangerous assumption, that the linked systems are always up. If this is for your tests, consider to switch to a scenario, that is fully under your control. Limit external dependencies

**Review by SebastianBezold — CHANGES_REQUESTED** [review]
Beside the project content check that i'll create, see my review comments for potential changes

**Comment by SebastianBezold** [comment]
Project content review: https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/9367

**Comment by fabiodmota** [comment]
> Hi @fabiodmota, since my last review you pushed new code to this PR. They are even annotated as feature. The PR is now of a size, that requires an IP check. I'll create the issue for that and link it here. Also, the `pom.xml` changed quite a bit and to me it seems like the spring-boot dependencies are new. The `DEPENDENCIES` file on the other hand hardly changed. Can you clarify that? Did you run dash? Do we need to open automated IP issues for the 3rd party dependencies?

Just updated DEPENDENCIES file , every dependency is approved so i suppose it will not need dependency IP check

**Review by SebastianBezold — APPROVED** [review]

**Comment by SebastianBezold** [comment]
Thx for the changes @fabiodmota !

**Pull request merged**